crates/walrus-core/src/encoding/quilt_encoding.rs

```diff
@@ -280,14 +644,105 @@ impl QuiltVersion for QuiltVersionV1 {
     }
 }
 
+/// The header of a encoded blob in QuiltVersionV1.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub struct BlobHeaderV1 {
+    pub length: u64, // the lower 34 bits are used for the length.
+    pub mask: u16,   // the lower 14 bits are used for the mask.
+}
```

Comment:
As discussed offline, I feel the choice of using 34 bits for the length causes a lot of additional code complexity with questionable benefits: Any blob above 4GiB should not be stored as part of a quilt anyway. My suggestion would be to just use 32 bits for the length and specify that a quilt can only contain blobs of at most 4GiB. cc @karlwuest